src/applications/disability-benefits/526EZ/reducer.js

```diff
@@ -1,6 +1,58 @@
+
+import _ from 'lodash/fp';
+
 import formConfig from './config/form';
 import { createSaveInProgressFormReducer } from '../../common/schemaform/save-in-progress/reducers';
 
+import {
+  SET_PRESTART_STATUS,
+  RESET_PRESTART_STATUS,
+  RESET_PRESTART_DISPLAY,
+  PRESTART_STATUSES
+} from './actions';
+
+const initialState = {
+  status: PRESTART_STATUSES.notAttempted,
+  data: undefined,
+  display: false
+};
+
+export const prestart = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_PRESTART_STATUS: {
+      const newState = _.set('status', action.status, state);
+      if (!state.display) {
+        newState.display = true;
+      }
+
+      if (action.data && state.data) {
+        const previousData = state.data;
+        newState.data = {};
+        newState.data.previous = previousData;
+        newState.data.current = action.data;
```

Comment:
What problem is this block solving?

Reply:
In the event that a user's ITF has expired, we display both the previous ITF's expiration date (when it expired), and the current expiration date (when the created ITF will expire).

Reply:
So the state at any one time can look like
```
state: { stuff }
```
or
```
state: {
  previous: { stuff }
  current: { stuff }
}
```
Is that right?

Can we move the previous data out of `newState.data` and into something like `newState.previousData` to keep the structure consistent?

Reply:
Yes, this is consistent with the other comments proposing to store currentExpirationDate and previousExpirationDate in data.